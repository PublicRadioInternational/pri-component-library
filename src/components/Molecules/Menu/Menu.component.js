/**
 * @file Menu.component.js
 * Exports a menu component.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './Menu.css';

/**
 * Component that renders a menu.
 */
const Menu = ({ menuItems }) => {
  const listItems = menuItems.map(listItem => (
    <li className={styles.menuItem} key={listItem.name}>
      <a className={styles.menuLink} href={listItem.url}>
        {listItem.name}
      </a>
    </li>
  ));

  return <ul className={styles.menu}>{listItems}</ul>;
};

Menu.propTypes = {
  menuItems: PropTypes.arrayOf(PropTypes.object)
};

Menu.defaultProps = {
  menuItems: {}
};

export default Menu;
