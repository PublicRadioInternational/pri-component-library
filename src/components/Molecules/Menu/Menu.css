/* import colors... */
@value colors: "../../00_global/colors.css";
@value grayLight from colors;

.menu {
  margin: 6px 0;
}

.menuitem {
  display: inline;
  padding: 0 0 0 6px;
  text-align: center;
}

.menuLink {
  color: grayLight;
  padding-right: 6px;
}
