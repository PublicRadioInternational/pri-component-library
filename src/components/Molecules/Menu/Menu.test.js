/**
 * @file Menu.test.js
 * Contains tests for Menu.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import Menu from './Menu.component';

describe('<Menu />', () => {
  it('Matches the Menu snapshot', () => {
    const component = renderer
      .create(
        <Menu
          menuItems={[
            { name: 'Google', url: 'http://google.com' },
            { name: 'Yahoo', url: 'http://yahoo.com' }
          ]}
        />
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
