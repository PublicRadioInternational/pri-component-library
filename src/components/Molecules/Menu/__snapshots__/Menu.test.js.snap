// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Matches the Menu snapshot 1`] = `
<ul
  className="menu"
>
  <li
    className="menuItem"
  >
    <a
      className="menuLink"
      href="http://google.com"
    >
      Google
    </a>
  </li>
  <li
    className="menuItem"
  >
    <a
      className="menuLink"
      href="http://yahoo.com"
    >
      Yahoo
    </a>
  </li>
</ul>
`;
